File: test/test/app/src/leanback/java/com/fongmi/android/tv/ui/presenter/ChannelPresenter.java
package com.fongmi.android.tv.ui.presenter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.leanback.widget.Presenter;

import com.bumptech.glide.Glide;
import com.fongmi.android.tv.bean.Channel;
import com.fongmi.android.tv.databinding.AdapterChannelBinding;

public class ChannelPresenter extends Presenter {

    private final OnClickListener listener;

    public ChannelPresenter(OnClickListener listener) {
        this.listener = listener;
    }

    public interface OnClickListener {

        void showEpg(Channel item);

        void onItemClick(Channel item);

        boolean onLongClick(Channel item);
    }

    @NonNull
    @Override
    public Presenter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent) {
        return new ViewHolder(AdapterChannelBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull Presenter.ViewHolder viewHolder, Object object) {
        Channel item = (Channel) object;
        ViewHolder holder = (ViewHolder) viewHolder;
        item.loadLogo(holder.binding.logo);
        holder.binding.name.setText(item.getName());
        holder.binding.number.setText(item.getNumber());
        holder.binding.getRoot().setSelected(item.isSelected());
        setOnClickListener(holder, view -> listener.onItemClick(item));
        holder.view.setOnLongClickListener(view -> listener.onLongClick(item));
        holder.binding.getRoot().setRightListener(() -> listener.showEpg(item));
    }

    @Override
    public void onUnbindViewHolder(@NonNull Presenter.ViewHolder viewHolder) {
        ViewHolder holder = (ViewHolder) viewHolder;
        Glide.with(holder.binding.logo).clear(holder.binding.logo);
    }

    public static class ViewHolder extends Presenter.ViewHolder {

        private final AdapterChannelBinding binding;

        public ViewHolder(@NonNull AdapterChannelBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/leafstudio/tvplayer/ui/ChannelPresenter.kt
package com.leafstudio.tvplayer.ui

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.leanback.widget.Presenter
import com.leafstudio.tvplayer.R
import com.leafstudio.tvplayer.model.Channel

/**
 * 频道卡片 Presenter
 * 用于在 Leanback UI 中展示频道项
 */
class ChannelPresenter : Presenter() {
    
    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_channel, parent, false)
        return ViewHolder(view)
    }
    
    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any) {
        val channel = item as Channel
        val cardView = viewHolder.view
        
        val titleView = cardView.findViewById<TextView>(R.id.channel_title)
        val groupView = cardView.findViewById<TextView>(R.id.channel_group)
        
        titleView.text = channel.name
        groupView.text = channel.group ?: "未分类"
        
        // 设置卡片尺寸
        cardView.layoutParams = ViewGroup.LayoutParams(300, 150)
    }
    
    override fun onUnbindViewHolder(viewHolder: ViewHolder) {
        // 清理资源
    }
}
